src/tests/class_tests/openms/source/MRMTransitionGroupPicker_test.cpp

```diff
@@ -454,6 +454,211 @@ START_SECTION((template < typename SpectrumT > void remove_overlapping_features(
+    2.955816667,2.964066667,2.97195,2.979833333,2.987716667,2.995616667,3.003516667,
+    3.011416667,3.01895,3.026833333,3.034366667,3.042266667,3.0498,3.05735,3.065233333,
+    3.073133333,3.080666667,3.0882,3.095733333,3.103633333,3.111533333,3.119066667,
+    3.126966667,3.134866667,3.14275,3.15065,3.15855,3.166433333,3.174333333,3.182233333,
+    3.190133333,3.198016667,3.205916667,3.213166667};
+  std::vector<double> intensity={1447,2139,1699,755,1258,1070,944,1258,1573,1636,
+    1762,1447,1133,1321,1762,1133,1447,2391,692,1636,2957,1321,1573,1196,1258,881,
+    1384,2076,1133,1699,1384,692,1636,1133,1573,1825,1510,2391,4342,10382,17618,
+    51093,153970,368094,632114,869730,962547,966489,845055,558746,417676,270942,
+    184865,101619,59776,44863,31587,24036,20450,20324,11074,9879,10508,7928,7110,
+    6733,6481,5726,6921,6670,5537,4971,4719,4782,5097,5789,4279,5411,4530,3524,
+    2139,3335,3083,4342,4279,3083,3649,4216,4216,3964,2957,2202,2391,2643,3524,
+    2328,2202,3649,2706,3020,3335,2580,2328,2894,3146,2769,2517};
+
+  RichPeakChromatogram chromatogram;
+  for (size_t k = 0; k < time.size(); k++)
+  {   
+    ChromatogramPeak peak;
+    peak.setMZ(time[k]);
+    peak.setIntensity(intensity[k]);
+    chromatogram.push_back(peak);
+  } 
+
+  // Features
+  double best_left = 2.477966667;
+  double best_right = 3.01895;
+  double peak_apex = 2.7045;
+
+  // Calculate peak apex
+  MRMTransitionGroupPicker picker;  
+
+  ConvexHull2D::PointArrayType hull_points;
+  double intensity_sum(0.0), rt_sum(0.0);
+  double peak_apex_int = -1;
+
+  double width_at_5 = 0.0;
+  double width_at_10 = 0.0;
+  double width_at_50 = 0.0;
+  double start_time_at_10 = 0.0;
+  double start_time_at_5 = 0.0;
+  double end_time_at_10 = 0.0;
+  double end_time_at_5 = 0.0;
+  double total_width = 0.0;
+  double tailing_factor = 0.0;
+  double asymmetry_factor = 0.0;
+  double baseline_delta_2_height = 0.0;
+  double slope_of_baseline = 0.0;
+  int points_across_baseline = 0;
+  int points_across_half_height = 0;
+
+  picker.calculatePeakApexInt_(chromatogram,
+    best_left,best_right,hull_points,
+    intensity_sum,
+    rt_sum,
+    peak_apex_int,
+    peak_apex);
+
+  TEST_REAL_SIMILAR(intensity_sum,6764562);
+  TEST_REAL_SIMILAR(rt_sum,151.890633338);
+  TEST_REAL_SIMILAR(peak_apex_int,966489);
+}
+END_SECTION
+
+START_SECTION(( void calculatePeakShapeMetrics_(const MSChromatogram& chromatogram, 
+  double best_left, double best_right, 
+  double peak_height, double peak_apex, double avg_noise_level,
+  double & width_at_5,
+  double & width_at_10,
+  double & width_at_50,
+  double & start_time_at_10,
+  double & start_time_at_5,
+  double & end_time_at_10,
+  double & end_time_at_5,
+  double & total_width,
+  double & tailing_factor,
+  double & asymmetry_factor,
+  double & baseline_delta_2_height,
+  double & slope_of_baseline,
+  int & points_across_baseline,
+  int & points_across_half_height) ))
+{
+  // Toy chromatogram
```

Comment:
can you give some information on this data? is it from real data or generated in R or python?
It is very important to know for tests if this is really ground truth data or only to detect regressions.

Reply:
I added some information: "data is taken from raw LC-MS/MS data points acquired for L-Glutamate in RBCs".  Let me know if more information is needed.